### PR TITLE
Added a NavgationBar example with nested Navigators

### PR DESCRIPTION
**Pull request by HansMuller**
Added an example of a NavigationBar where each destination is managed by a nested Navigator.

Try the example in DartPad: https://dartpad.dev/?id=37a8f13c000cf229857f6c36b4a55ee6

Addresses issues raised in https://github.com/flutter/flutter/issues/18740. Similar examples to come for NavigationRail, BottomNavigationBar, and TabBar.

### CODE REVIEW

File: examples/api/lib/material/navigation_bar/navigation_bar.0.dart [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+              final Widget view = destinationViews[index];
+              if (index == selectedIndex) {
+                destinationFaders[index].forward();
+                return view;
+              } else {
+                destinationFaders[index].reverse();
+                if (destinationFaders[index].isAnimating) {
+                  return IgnorePointer(child: view);
+                }
+                return Offstage(child: view);
```
Comment (chunhtai) [inline examples/api/lib/material/navigation_bar/navigation_bar.0.dart:344]: We can wrap all child with Offstage and toggle the flag instead

File: examples/api/lib/material/navigation_bar/navigation_bar.0.dart [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+  const RootPage({ Key? key, required this.destination }) : super(key: key);
+
+  final Destination destination;
+
+  Widget _buildDialog(BuildContext context) {
+    return AlertDialog(
+      title: Text('${destination.title} AlertDialog'),
+      actions: <Widget>[
+        TextButton(
+          onPressed: () { Navigator.pop(context, 'OK'); },
```
Comment (chunhtai) [inline examples/api/lib/material/navigation_bar/navigation_bar.0.dart:27]: what is the result for?

**Review by chunhtai — COMMENTED** [review]
Mostly looks good, left some comments

File: examples/api/lib/material/navigation_bar/navigation_bar.0.dart (hunk 2 above)
Comment (HansMuller) [reply examples/api/lib/material/navigation_bar/navigation_bar.0.dart:27]: Good point, there's no need for the 'OK' argument.

File: examples/api/lib/material/navigation_bar/navigation_bar.0.dart (hunk 1 above)
Comment (HansMuller) [reply examples/api/lib/material/navigation_bar/navigation_bar.0.dart:344]: Good idea, that way the tree doesn't need to change.

File: examples/api/lib/material/navigation_bar/navigation_bar.0.dart [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+  const RootPage({ Key? key, required this.destination }) : super(key: key);
+
+  final Destination destination;
+
+  Widget _buildDialog(BuildContext context) {
+    return AlertDialog(
+      title: Text('${destination.title} AlertDialog'),
+      actions: <Widget>[
+        TextButton(
+          onPressed: () { Navigator.pop(context); },
```
Comment (chunhtai) [inline examples/api/lib/material/navigation_bar/navigation_bar.0.dart:27]: This can be arrow method

**Review by chunhtai — APPROVED** [review]
LGTM

File: examples/api/lib/material/navigation_bar/navigation_bar.0.dart (hunk 3 above)
Comment (HansMuller) [reply examples/api/lib/material/navigation_bar/navigation_bar.0.dart:27]: Personally, I prefer to only use the `=>` notation for functions that return a value.

**Comment by mark8044** [comment]
@HansMuller 
@chunhtai 

When running this on iOS, scroll to top functionality doesn't work when tapping the status bar

**Pull request merged**